dev/com.ibm.ws.grpc_fat/fat/src/com/ibm/ws/fat/grpc/ClientHeaderPropagationTests.java

```diff
@@ -0,0 +1,319 @@
+            webClient.addRequestHeader("testHeader1", "456");
+            webClient.addRequestHeader("testHeader2", "789");
+
+            // Log the page for debugging if necessary in the future.
+            Log.info(c, name.getMethodName(), page.asText());
+            Log.info(c, name.getMethodName(), page.asXml());
+
+            assertTrue("the servlet was not loaded correctly",
+                       page.asText().contains("gRPC helloworld client example"));
+
+            HtmlForm form = page.getFormByName("form1");
+
+            // set a name in the form, which we'll expect the RPC to return
+            HtmlTextInput inputText = (HtmlTextInput) form.getInputByName("user");
+            inputText.setValueAttribute("us3r1");
+
+            // set the port of the grpcserver in the form
+            HtmlTextInput inputPort = (HtmlTextInput) form.getInputByName("port");
+            inputPort.setValueAttribute(String.valueOf(GrpcServer.getHttpDefaultPort()));
+
+            // set the hostname of the gprcserver in the form
+            HtmlTextInput inputHost = (HtmlTextInput) form.getInputByName("address");
+            inputHost.setValueAttribute(GrpcServer.getHostname());
+
+            // submit to the grpcClient, and execute the RPC
+            HtmlSubmitInput submitButton = form.getInputByName("submit");
+            page = submitButton.click();
+
+            // Log the page for debugging if necessary in the future.
+            Log.info(c, name.getMethodName(), page.asText());
+            assertTrue("the gRPC request did not complete correctly", page.asText().contains("us3r1"));
+
+            //Make sure the Interceptor was called to verify match
+            String interceptorHasRun = GrpcServer.waitForStringInLog("com.ibm.ws.grpc.fat.helloworld.service.HelloWorldServerInterceptor3 has been invoked!",
+                                                                     SHORT_TIMEOUT);
+            if (interceptorHasRun == null) {
+                Assert.fail(c + ": server.xml with <grpcTarget> element: no interceptor ran when it should have in " + SHORT_TIMEOUT + "ms");
+            }
+
+            // make sure expected headers were found
+            String headerFound = GrpcServer.waitForStringInLog("testHeader1=456,testHeader2=789", SHORT_TIMEOUT);
+            if (headerFound == null) {
+                Assert.fail(c + ": testHeader1 or testHeader2 not found when it should have in " + SHORT_TIMEOUT + "ms");
+            }
+            //Make sure the testHeader is not displayed by the Interceptor
+            headerFound = GrpcServer.verifyStringNotInLogUsingMark("testHeader=123", SHORT_TIMEOUT);
+            if (headerFound != null) {
+                Assert.fail(c + ": testHeader found when it should not have");
+            }
+        }
+    }
+
+    /**
+     * This method is used to set the server.xml
+     *
+     * @throws Exception
+     */
+    private static void setServerConfiguration(LibertyServer server,
```

Comment:
There's a commong setServerconfig now in the grpcUtils class for all the tests to use.

Reply:
change to use GrpcTestUtils.setServerConfiguration